### PR TITLE
build: fix peer depends and update common depends

### PR DESCRIPTION
**Pull request by ghassanmas**
fix peer depends and update common depends

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ghassanmas!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@ghassanmas Thank you for your contribution, I kicked off the checks.

**Comment by ghassanmas** [comment]
@natabene Tests should be resolved now I have run them on the fork. Also apply for other PRs in profile and gradebook

**Comment by timmc-edx** [comment]
@davidjoy Would you be able to review this one, or point me to who should?

File: package.json [hunk 1]
```diff
@@ -30,18 +30,18 @@
...
+    "@edx/frontend-component-header": "2.4.6",
+    "@edx/frontend-platform": "1.15.6",
+    "@edx/paragon": "19.14.1",
     "@fortawesome/fontawesome-svg-core": "1.2.36",
     "@fortawesome/free-brands-svg-icons": "5.15.4",
     "@fortawesome/free-regular-svg-icons": "5.15.4",
     "@fortawesome/free-solid-svg-icons": "5.15.4",
     "@fortawesome/react-fontawesome": "0.1.16",
     "@tensorflow-models/blazeface": "0.0.7",
-    "@tensorflow/tfjs-converter": "1.7.4",
```
Comment (davidjoy) [inline package.json:43]: This bumps these tensorflow libs through two major versions - do we know that we've absorbed the breaking changes in them and that this won't break?

**Review by davidjoy — CHANGES_REQUESTED** [review]
One change request and one sanity check on some of the version bumps.

File: package.json (hunk 1 above)
Comment (ghassanmas) [reply package.json:43]: This patch more or less a backport of https://github.com/openedx/frontend-app-account/commit/0527c735295949c71074021b07530d2c70e38cb0  and more of discussion can be found https://github.com/openedx/frontend-app-account/pull/611

File: package.json [hunk 2]
```diff
@@ -91,5 +91,8 @@
     "react-test-renderer": "16.14.0",
     "reactifex": "1.1.1",
     "redux-mock-store": "1.5.4"
+  },
+    "peerDependencies": {
```
[the remark this answers is not in the document]
Comment (ghassanmas) [reply package.json:95]: Anyway if you checkout to fronted-app-learning amd run `npm ls boostrap` you should get 4.6.1.

File: package.json (hunk 1 above)
Comment (ghassanmas) [reply package.json:43]: These are not really used directly, I have researched the code and they are just there as peer depends of `@tensorflow-models/blazeface 0.0.7` which require those exact versions [hence](https://github.com/tensorflow/tfjs-models/blob/84a4d132f120bcb24cfbc2b16ede7956e5ea9588/blazeface/package.json#L16-L17). That being said, if something would break it would more likely to have broken before.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidjoy) [reply package.json:95]: MFEs can't have peerDependencies - they're not a package that gets installed as a dependency.  I don't know if somehow adding peerDependencies removes a warning or something, but I don't think that's the right way to solve whatever problem we're seeing.

It should not matter if MFEs have consistent versions of paragon - the code isn't shared anyway, I don't think it's meaningful to optimize for a consistent version.  We shouldn't limit an MFEs version of a library because a different, independent MFE is at a particular version, that's not how the system was designed/intended to be used.  

If we need to update these dependencies for nutmeg for some reason, we should just update them to latest - artificially pinning them at earlier versions is only going to cause problems and confusion later.

[the remark this answers is not in the document]
Comment (ghassanmas) [reply package.json:95]: > MFEs can't have peerDependencies - they're not a package that gets installed as a dependency. I don't know if somehow adding peerDependencies removes a warning or something, but I don't think that's the right way to solve whatever problem we're seeing.


The end goal is to force to use a spesfic verion of bootsrap, and I force to do so because otherwise it would break the build. I think it would be doable to have the same result if I instead moved it to deps instead of creating per depnds. 

>  It should not matter if MFEs have consistent versions of paragon - the code isn't shared anyway, I don't think it's meaningful to optimize for a consistent version. We shouldn't limit an MFEs version of a library because a different, independent MFE is at a particular version, that's not how the system was designed/intended to be used.

Again the pinned vresion would resolve the failed build because the artificial created version of boostrap after updarading paragon will result in failed build. When I was debuggin this, I was confused, such that why the paragon update made the build to fail while the same version is being in learning and it builds okay. This made me further debugging the depends of paragon between the learning and other MFEs what I found is that boostrap version is different in the learning thus I tried pin  boostrap version, rebuild again and the result was ok. 

>  If we need to update these dependencies for nutmeg for some reason, we should just update them to latest - artificially pinning them at earlier versions is only going to cause problems and confusion later.

I agree that the right path would be frequently update depends for each MFE (Note: I am referring to the open release branches). The thing is its hard to know which version suppose to be compitable with open-release, so my stratgey is find the maximum version of all MFEs and consider that the right version.

Add to that for paragon for example the version is already pinned so artificially updaing the depends is not possible, a different startgey would be to go check the change-log of paragon version 19.* and find if there is any breaking changes and if not, then use the last minor version of paragon 19. which in this case is [19.25.3](https://github.com/openedx/paragon/releases/tag/v19.25.3)

So in summary; _Assuming using paragon 19.25.3 wouldn't require us to pin boostrap and assuming MFEs would work as epxected using that version_. Would you accept such change? (To MFEs nutmeg branch, learning, account profile and gradebook)

[the remark this answers is not in the document]
Comment (ghassanmas) [reply package.json:95]: @davidjoy even using the latest minor version of paragon 19 didn't resolve it. 
Anyway I stumbled upon this issue while checking paragon openedx/paragon/pull/1562 it seems the way to resolve to include the new varriable of bootsrap and if an MFE is using older paragon the solution was to pin it just as I did here [ref commit in paryment mfe](https://github.com/openedx/frontend-app-payment/commit/cab00beedc3ee6a8cf5fe809d04ede30456c6d42) The difference is that they don't add it per rather direclty in depends. 

However I went over npm docs, and the right way to do is to define [overrides](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) 
I am now testing with overrides and see if build would  success and then I willl deploy the same version in production to see if it would also would as expected

[the remark this answers is not in the document]
Comment (ghassanmas) [reply package.json]: I tested and worked as expcted. 
To test in production you can go: [https://apps.edx.gsgapp.io/account](https://apps.edx.gsgapp.io/account)  you can use admin@zaat.dev and "zaatdev" as pass.

**Comment by abdullahwaheed** [comment]
@ghassanmas there are some snapshots failing and local development server is not working with following error:
```
> @edx/frontend-app-account@1.0.0-semantically-released start
> fedx-scripts webpack-dev-server --progress

Running with resolved config:
/.../frontend-app-account/node_modules/@edx/frontend-build/config/webpack.dev.config.js

No local module configuration file found. This is fine.
1% setup initialize[webpack-cli] TypeError: cli.isMultipleCompiler is not a function
    at Command.<anonymous> (.../frontend-app-account/node_modules/@webpack-cli/serve/lib/index.js:146:35)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Promise.all (index 1)
    at async Command.<anonymous> (.../frontend-app-account/node_modules/webpack-cli/lib/webpack-cli.js:1672:7)

```

 Please pull latest code from master, this error is due to old version of webpack, which is being updated in frontend-build.

**Comment by ghassanmas** [comment]
@abdullahwaheed just pushed a fix, by overrideing webpack-cli version hence webpack/webpack/issues/15951 . Also note this error already exists in nutmeg.master.

**Comment by ghassanmas** [comment]
Secondaly for the tests I have backported 0527c735295949c71074021b07530d2c70e38cb0 and seems to resolve the failing tests

**Comment by BilalQamar95** [comment]
@ghassanmas Are we sure that `useEffect` dependencies update won't cause unnecessary renders? If this was just to satisfy the linter I'd probably revert this and leave it as-is, ignoring the rule in case of `useEffect` Hook.

**Comment by ghassanmas** [comment]
> @ghassanmas Are we sure that `useEffect` dependencies update won't cause unnecessary renders? If this was just to satisfy the linter I'd probably revert this and leave it as-is, ignoring the rule in case of `useEffect` Hook.

That chagne is backported to from https://github.com/openedx/frontend-app-account/commit/0527c735295949c71074021b07530d2c70e38cb0 to fix the failed tests.

**Comment by arbrandes** [comment]
Reviewers, how are we feeling about this?  Any further changes requested?  (@davidjoy, do you want to hand off final say to somebody else?  Who owns the Account MFE, these days?)

**Comment by arbrandes** [comment]
Also, @ghassanmas, do you mind commenting on https://github.com/openedx/frontend-wg/issues/140 so I can assign it to you?  Or maybe you can assign yourself, not sure. :)

**Comment by ghassanmas** [comment]
@arbrandes I don't I can, but I have just comment there my perspective/finding  on that.

**Comment by mphilbrick211** [comment]
Hi @arbrandes and @ghassanmas! Just checking to see if this is still being pursued? If so, can it be reviewed/merged?

**Comment by mphilbrick211** [comment]
@ghassanmas @arbrandes friendly ping on this :)

**Comment by ghassanmas** [comment]
@mphilbrick211 I think it's fine to close it, since I don't think we support nutmeg releases anymore.

**Comment by mphilbrick211** [comment]
Ok, closing!

**Comment by openedx-webhooks** [comment]
@ghassanmas Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-account&entry.867055334=https://github.com/openedx/frontend-app-account/pull/615&entry.1484655318=https://github.com/ghassanmas&entry.752974735=2022-07-31+08:22&entry.1917517419=2023-02-08+20:51&entry.2133058324=No) so we can improve your experience in the future.
